Terminal.Gui/Drawing/Region.cs

```diff
@@ -73,8 +77,10 @@ public override bool Equals (object? obj)
     /// </summary>
     /// <param name="regions"></param>
     /// <returns></returns>
-    public static dynamic Union (params Region [] regions)
+    public static dynamic? Union (params Region [] regions)
```

Comment:
Please do not use `dynamic` for anything, anywhere.

It's highly reflection-dependent and has a ***massive*** runtime cost that is almost certainly a lot higher than what this is trying to achieve in the first place.

Reply:
params argument arrays also force boxing.

This isn't appropriate for the goal of this.

Reply:
> Please do not use `dynamic` for anything, anywhere.
> 
> It's highly reflection-dependent and has a _**massive**_ runtime cost that is almost certainly a lot higher than what this is trying to achieve in the first place.

The use of `dynamic` was massively considered because I wanted something that could only return a nullable of `Rectangle` and `RectangleF`, reusing code without duplication. I still thought about using generic type (`T`) but that would imply a big change in the API to change to `Region<Rectangle>`, `Region<RectangleF>`. Could the methods I created return null, if applicable, and if I used a `Rectangle?` nullable or `RectangleF`? return, it would also imply a major change to the entire code. Therefore, I chose to use `dynamic` because it can return any nullable of any of the two types I mentioned, `Rectangle`, `RectangleF`, without major code changes and without code duplication. Therefore, in my opinion it will be worth the sacrifice.

Reply:
> params argument arrays also force boxing.
> 
> This isn't appropriate for the goal of this.

You're right and it doesn't even make sense to use `params` because the parameter is already an array, so I did something wrong and will remove it. Thanks.